src/main.cpp

```diff
@@ -55,40 +55,52 @@ class any_error_reporter {
     std::visit(
         [&](auto &r) {
           using reporter_type = std::decay_t<decltype(r)>;
-          if constexpr (std::is_base_of_v<vim_qflist_json_error_reporter,
+          if constexpr (std::is_base_of_v<error_tape<vim_qflist_json_error_reporter>,
                                           reporter_type>) {
-            r.set_source(input, file.path, file.vim_bufnr);
+            r.get_reporter()->set_source(input, file.path, file.vim_bufnr);
           } else {
-            r.set_source(input, file.path);
+            r.get_reporter()->set_source(input, file.path);
           }
         },
-        this->reporter_);
+        this->tape_);
   }
 
   error_reporter *get() noexcept {
-    return std::visit([](error_reporter &r) { return &r; }, this->reporter_);
+    try {
+      return &std::get<error_tape<text_error_reporter>>(this->tape_);
+    } catch(std::bad_variant_access&) {
+      return &std::get<error_tape<vim_qflist_json_error_reporter>>(this->tape_);
+    }
```

Comment:
Why don't we use `std::visit` like with the original code?
```suggestion
    return std::visit([](error_reporter &r) { return &r; }, this->tape_);
```

Reply:
This is normally corrected.